src/main/webapp/app/overview/course-statistics/course-statistics.component.html

```diff
@@ -80,34 +73,83 @@ <h2 class="text-center">{{ 'artemisApp.courseOverview.statistics.presentationSco
+                                    }}
+                                </span>
+                                <span *ngIf="model.name === barChartTitle.BONUS">
+                                    <span>| {{ 'artemisApp.courseOverview.statistics.bonusPointTooltip' | artemisTranslate }}</span>
+                                </span>
+                                <span *ngIf="model.name === barChartTitle.NOT_INCLUDED">
+                                    <span>| {{ 'artemisApp.courseOverview.statistics.notIncludedTooltip' | artemisTranslate }}</span>
+                                </span>
+                            </div>
+                            <div [ngSwitch]="model.name">
+                                <div *ngSwitchCase="barChartTitle.MISSED">
+                                    <span *ngIf="!model.afterDueDate && !model.notParticipated && !model.isProgrammingExercise">
+                                        {{
+                                            'artemisApp.courseOverview.statistics.exerciseMissedScore'
+                                                | artemisTranslate: { points: model.absoluteValue, percentage: roundScoreSpecifiedByCourseSettings(model.value, course) }
+                                        }}
+                                    </span>
+                                    <span *ngIf="model.notParticipated">
+                                        {{ 'artemisApp.courseOverview.statistics.exerciseNotParticipated' | artemisTranslate: { exercise: model.exerciseTitle } }}
+                                    </span>
+                                    <span *ngIf="model.afterDueDate">
+                                        {{ 'artemisApp.courseOverview.statistics.exerciseParticipatedAfterDueDate' | artemisTranslate: { exercise: model.exerciseTitle } }}
+                                    </span>
+                                    <span *ngIf="model.isProgrammingExercise && !model.notParticipated">
+                                        {{
+                                            'artemisApp.courseOverview.statistics.programmingExerciseFailedTests'
+                                                | artemisTranslate: { tests: model.absoluteValue, percentage: roundScoreSpecifiedByCourseSettings(model.value, course) }
+                                        }}
+                                    </span>
+                                </div>
+                                <div *ngSwitchCase="barChartTitle.NOT_GRADED">
+                                    <span> {{ 'artemisApp.courseOverview.statistics.exerciseNotGraded' | artemisTranslate: { exercise: model.exerciseTitle } }}</span>
+                                </div>
+                            </div>
+                        </ng-template>
+                    </ngx-charts-bar-horizontal-stacked>
                 </div>
                 <div class="col-12 col-md-4 align-self-center" *ngIf="exerciseGroup">
```

Comment:
Since `exerciseGroup` is now an array, maybe we can also check its length to be non-zero here if there is a possibility that it can be zero.

Reply:
Actually, only non-empty arrays should be passed to the framework but I added a check in a5044d3 just to make sure.